classes/Cart.php

```diff
@@ -534,8 +534,10 @@ public function getCartRules($filter = CartRule::FILTER_ACTION_ALL, $autoAdd = t
             $row['value_real'] = $row['obj']->getContextualValue(true, $virtual_context, $filter);
             $row['value_tax_exc'] = $row['obj']->getContextualValue(false, $virtual_context, $filter);
             // Retro compatibility < 1.5.0.2
-            $row['id_discount'] = $row['id_cart_rule'];
-            $row['description'] = $row['name'];
+            if(Tools::version_compare(_PS_VERSION_, '1.5.0.2', '<') === true){
```

Comment:
Missing a space between if and the (.

Also, we will never been in this if statement.